sources/SilkTouch/SilkTouch/Mods/TransformCOM.cs

```diff
@@ -0,0 +1,749 @@
+                    return base.VisitVariableDeclaration(node);
+
+                // Get the corresponding parameter type
+                var variableType = node.Type;
+                var name = GetTypeName(variableType?.ToString() ?? string.Empty);
+
+                // Check if the parameter type is ComType
+                if (!ComTypes.Any(type => $"{type.Item1}*" == name))
+                {
+                    return base.VisitVariableDeclaration(node);
+                }
+
+                var newInstantiation = node;
+                var variables = node.Variables;
+                for (int i = 0; i < variables.Count; i++)
+                {
+                    var variableExpression = variables[i].Initializer?.Value as LiteralExpressionSyntax;
+                    // Check if the variable is instantiated to a null literal
+                    if (variableExpression is not null && variableExpression.IsKind(SyntaxKind.NullLiteralExpression))
+                    {
+                        var newDefaultArg = LiteralExpression(SyntaxKind.DefaultLiteralExpression);
+                        newInstantiation = newInstantiation.ReplaceNode(variableExpression, newDefaultArg);
+                    }
+                }
+                return newInstantiation;
+            }
+
+            private string GetTypeName(string fullTypeName)
+            {
+                int dot = fullTypeName.LastIndexOf('.');
+
+                if (dot != -1)
+                {
+                    return fullTypeName.Substring(dot + 1);
+                }
+
+                return fullTypeName;
+            }
+        }
+
+        class Rewriter(List<(string, bool, KeyedStringTree?)> ComTypes)
+            : CSharpSyntaxRewriter
+        {
+            public override SyntaxNode? VisitPointerType(PointerTypeSyntax node)
+            {
+                for (int i = 0; i < ComTypes.Count; i++)
+                {
+                    (string, bool, KeyedStringTree?) val = ComTypes[i];
+
+                    if (val.Item1 == node.ElementType.ToString())
+                    {
+                        return IdentifierName(val.Item1);
+                    }
+                }
+
+                return base.VisitPointerType(node);
+            }
+
+            public override SyntaxNode? VisitIdentifierName(IdentifierNameSyntax node)
+            {
+                if (node.Identifier.ToString() == "lpVtbl")
+                {
+                    return ParenthesizedExpression(PrefixUnaryExpression(SyntaxKind.PointerIndirectionExpression, node));
+                }
+
+                return base.VisitIdentifierName(node);
+            }
+
+            public override SyntaxNode? VisitVariableDeclaration(VariableDeclarationSyntax node)
+            {
+                if (node.Type.ToString() == "void**" && node.Variables.First().Identifier.ToString() == "lpVtbl")
```

Comment:
What about the other variables? (If any ofc)

Comes back to my previous comment about maybe determining the pointer field from the type itself, afaik the only "is this COM" check we're doing is looking for the presence of a nested interface. We should probably check that the struct looks how we expect it to as well.

Reply:
Are you concerned about a case where a COM interface has an additional variable? because I don't think that ever happens as I believe that would violate it being COM. If we are thinking about a class that inherits from a COM class but isn't COM, I also don't think that happens, but this would be a problem.
We could check that only one variable is declared in our COM check and that it is only a void**, but again I believe for this case to come up would cause other issues with COM objects since they are pointers to shared objects afaik.

Reply:
doing that check now